### PR TITLE
Orca: add py38 ray example test

### PR DESCRIPTION
**Pull request by hamham223**
### 1. Why the change?

https://github.com/analytics-zoo/arda-docker/issues/773#issuecomment-1432651878
~The last python 3.8 test. Orca can support python 3.8!!!~
Example-Tests are all added with python 3.8, and tested in nightly only (cost too much time).

Next step: add python 3.8 test for orca tutorial

### 2. Summary of the change 

+ add `orca-exampletest-ray-py38-spark3` (and nightly)
+ change job order in `PR_validation.yml` (basic, pytest, tutorial, ~example~)

### 4. How to test?
- [x] Path Filter
- [x] Orca-Exampletest-Ray-Py38-Spark3
- [x] Orca-Exampletest-Ray-Py38-Spark3 (Nightly)

### CODE REVIEW

**Comment by hamham223** [comment]
Remove the example-test from PR? cost about 50 min

File: python/orca/dev/example/run-example-test-ray.sh [hunk 1]
```diff
@@ -52,6 +52,9 @@ python ${BIGDL_ROOT}/python/orca/example/ray_on_spark/rllib/multi_agent_two_trai
 now=$(date "+%s")
 time5=$((now-start))
 
+python_version=$(python --version | awk '{print$2}')
+if [ $python_version == 3.7.10 ];then
```
Comment (hamham223) [inline python/orca/dev/example/run-example-test-ray.sh:56]: We specify the python version when [setup-env](https://github.com/intel-analytics/BigDL/blob/main/.github/actions/orca/setup-env/setup-orca-python-py37-basic/action.yml#L9).

File: .github/actions/orca/orca-exampletest-ray-py38-spark3-action/nightly-test/action.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        pip install -i ${GONDOLIN_PIP_MIRROR} --trusted-host ${GONDOLIN_TRUSTED_HOST} tqdm==4.46.1
+        pip install -i ${GONDOLIN_PIP_MIRROR} --trusted-host ${GONDOLIN_TRUSTED_HOST} dm_tree==0.1.8
+        pip install -i ${GONDOLIN_PIP_MIRROR} --trusted-host ${GONDOLIN_TRUSTED_HOST} mxnet==1.6.0
+        pip install -i ${GONDOLIN_PIP_MIRROR} --trusted-host ${GONDOLIN_TRUSTED_HOST} matplotlib==3.5.3
+        pip install -i ${GONDOLIN_PIP_MIRROR} --trusted-host ${GONDOLIN_TRUSTED_HOST} pyarrow==4.0.1
+        pip install -i ${GONDOLIN_PIP_MIRROR} --trusted-host ${GONDOLIN_TRUSTED_HOST} albumentations
+        pip install -i ${GONDOLIN_PIP_MIRROR} --trusted-host ${GONDOLIN_TRUSTED_HOST} grpcio==1.43.0
+
+        # mxnet related test does not support numpy 1.24
+        pip install -i ${GONDOLIN_PIP_MIRROR} --trusted-host ${GONDOLIN_TRUSTED_HOST} numpy==1.21.6
```
Comment (hamham223) [inline .github/actions/orca/orca-exampletest-ray-py38-spark3-action/nightly-test/action.yml:31]: @hkvision Should I skip this test in the shell scripts or install a lower version numpy? The ut passed in np 1.21.6

File: python/orca/dev/example/run-example-tests.sh [hunk 3]
```diff
@@ -137,20 +137,7 @@ python ${BIGDL_ROOT}/python/orca/example/freeze_saved_model/freeze.py \
...
-    -P tmp/data/NAB/nyc_taxi/
-fi
-#timer
-start=$(date "+%s")
-python ${BIGDL_ROOT}/python/orca/example/data/spark_pandas.py \
-  -f tmp/data/NAB/nyc_taxi/nyc_taxi.csv
-
-now=$(date "+%s")
-time4=$((now - start))
+# #4 orca data test is tested in ray example test
```
Comment (hkvision) [inline python/orca/dev/example/run-example-tests.sh:140]: do not add this remark

File: python/orca/dev/example/run-example-tests.sh [hunk 4]
```diff
@@ -245,7 +232,6 @@ time9=$((now - start))
 echo "#1 openvino time used: $time1 seconds"
 echo "#2 vnni/openvino time used: $time2 seconds"
 echo "#3 tensorflow time used: $time3 seconds"
-echo "#4 orca data time used:$time4 seconds"
 echo "#5 orca tf imagesegmentation time used:$time5 seconds"
 echo "#6 orca tf transfer_learning time used:$time6 seconds"
 echo "#7 orca tf basic_text_classification time used:$time7 seconds"
```
Comment (hkvision) [inline python/orca/dev/example/run-example-tests.sh:237]: change the numberings.

File: python/orca/dev/example/run-example-test-ray.sh [hunk 5]
```diff
@@ -52,6 +52,9 @@ python ${BIGDL_ROOT}/python/orca/example/ray_on_spark/rllib/multi_agent_two_trai
 now=$(date "+%s")
 time5=$((now-start))
 
+python_version=$(python --version | awk '{print$2}')
```
Comment (hkvision) [inline python/orca/dev/example/run-example-test-ray.sh:55]: put this get python version to the beginning of the file?

File: python/orca/dev/example/run-example-test-ray.sh [hunk 6]
```diff
@@ -63,6 +66,7 @@ start=$(date "+%s")
 python ${BIGDL_ROOT}/python/orca/example/ray_on_spark/parameter_server/sync_parameter_server.py --iterations 10
 now=$(date "+%s")
 time7=$((now-start))
+fi
```
Comment (hkvision) [inline python/orca/dev/example/run-example-test-ray.sh:69]: will it throw error when time6 and time7 are not defined?

Comment (hamham223) [reply python/orca/dev/example/run-example-test-ray.sh:69]: no, just output nothing

File: .github/workflows/PR_validation.yml [hunk 7]
```diff
@@ -498,12 +437,12 @@ jobs:
...
       uses: ./.github/actions/orca/setup-env/setup-orca-python-py37-basic
     - name: Run Test
-      uses: ./.github/actions/orca/orca-tutorial-ncf-ray-py37-spark3-action/pr-test
+      uses: ./.github/actions/orca/orca-pytest-openvino-py37-spark3
     - name: Remove Env
       if: ${{ always() }}
       uses: ./.github/actions/remove-env
 
-  Orca-Tutorial-NCF-Py37-Spark3:
+  Orca-Tutorial-Xshards-Py37-Spark3:
```
Comment (hkvision) [inline .github/workflows/PR_validation.yml:445]: this should be removed from PR?

File: python/orca/dev/example/run-example-tests.sh [hunk 8]
```diff
@@ -178,7 +163,7 @@ fi
 now=$(date "+%s")
 time5=$((now - start))
```
Comment (hkvision) [inline python/orca/dev/example/run-example-tests.sh:164]: also need to modify time

Comment (hamham223) [reply python/orca/dev/example/run-example-tests.sh:164]: 😢

File: .github/actions/orca/orca-exampletest-ray-py38-spark3-action/nightly-test/action.yml (hunk 2 above)
Comment (hamham223) [reply .github/actions/orca/orca-exampletest-ray-py38-spark3-action/nightly-test/action.yml]: good try, seems use np 1.24 still cause some problems

Comment (hamham223) [reply .github/actions/orca/orca-exampletest-ray-py38-spark3-action/nightly-test/action.yml]: https://github.com/ray-project/ray/issues/31565 seems rlib does not support numpy 1.24, need to install the nightly version

**Comment by hamham223** [comment]
Seems scikit-learn==0.22.2.post1 does not support numpy 1.24

##### How to reproduce
env: python 3.8

```python
pip install scikit-learn==0.22.2.post1
pip install albumentations
```

Test scripts:
```python
from albumentations import *
print("success")
```

**Comment by hkvision** [comment]
can we use other sklearn versions?

**Comment by hamham223** [comment]
> can we use other sklearn versions?

install without specify the version?
```
pip install -i ${GONDOLIN_PIP_MIRROR} --trusted-host ${GONDOLIN_TRUSTED_HOST} scikit-learn
```
or just remove this line, since `albumentations` automatically installs scikit-learn

**Comment by hkvision** [comment]
> > can we use other sklearn versions?
> 
> install without specifying the version?
> 
> ```
> pip install -i ${GONDOLIN_PIP_MIRROR} --trusted-host ${GONDOLIN_TRUSTED_HOST} scikit-learn
> ```
> 
> or just remove this line, since `albumentations` automatically installs scikit-learn

We can do this.

**Pull request merged**